src/libAtomVM/opcodesswitch.h

```diff
@@ -4970,6 +4992,105 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
             }
 #endif
 
+#ifdef ENABLE_OTP24
+            case OP_MAKE_FUN3: {
+                int next_off = 1;
+                int fun_index;
+                DECODE_LABEL(fun_index, code, i, next_off, next_off);
+                dreg_t dreg;
+                dreg_type_t dreg_type;
+                DECODE_DEST_REGISTER(dreg, dreg_type, code, i, next_off, next_off);
+
+                next_off++; // skip extended list tag
+
+                #ifdef IMPL_EXECUTE_LOOP
+                    uint32_t n_freeze = module_get_fun_freeze(mod, fun_index);
+
+                    int size = 2 + n_freeze;
+                    if (memory_ensure_free(ctx, size + 1) != MEMORY_GC_OK) {
+                        RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+                    }
```

Comment:
it took me a moment to realize that it was size + 3, are we basically doing `(2 + n_freeze) + 1` for any readability reason? Anyway it might be also a good idea to #define a constant so we end having less hardcoded stuff.

Reply:
Thanks. Simplified via using the `size` value from the instruction rather than getting it from `module_get_fun_freeze`.